Sources/Controllers/DashboardOnMap/MapSettings/MapSettingsGpxViewController.swift

```diff
@@ -0,0 +1,592 @@
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.tableView.setEditing(true, animated: false)
+        self.tableView.allowsMultipleSelectionDuringEditing = true
+        self.navigationItem.hidesSearchBarWhenScrolling = true
+        self.definesPresentationContext = true
+        updateSelectedRows()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        if let searchController = searchController, searchController.isActive {
+            searchController.isActive = false
+            self.navigationItem.searchController = nil
+        }
+    }
+    
+    override func getTitle() -> String? {
+        localizedString("shared_string_gpx_tracks")
+    }
+    
+    override func getLeftNavbarButtonTitle() -> String? {
+        localizedString("shared_string_cancel")
+    }
+    
+    override func getRightNavbarButtons() -> [UIBarButtonItem]? {
+        guard let searchButton = createRightNavbarButton(nil, iconName: "ic_navbar_search", action: #selector(onSearchButtonClicked), menu: nil),
+              let importButton = createRightNavbarButton(nil, iconName: "ic_custom_import_outlined", action: #selector(onImportButtonClicked), menu: nil) else { return nil }
+        
+        searchButton.accessibilityLabel = localizedString("shared_string_search")
+        importButton.accessibilityLabel = localizedString("import_tracks")
+        return [importButton, searchButton]
+    }
+    
+    override func createSubview() -> UIView? {
+        if !isSearchActive {
+            segmentedControl = UISegmentedControl(items: [localizedString("shared_string_visible"), localizedString("shared_string_all")])
+            segmentedControl?.selectedSegmentIndex = 0
+            isShowingVisibleTracks = true
+            segmentedControl?.addTarget(self, action: #selector(segmentChanged(_:)), for: .valueChanged)
+            return segmentedControl
+        } else {
+            return nil
+        }
+    }
+    
+    override func hideFirstHeader() -> Bool {
+        true
+    }
+    
+    override func getBottomAxisMode() -> NSLayoutConstraint.Axis {
+        .horizontal
+    }
+    
+    override func getTopButtonTitle() -> String? {
+        if !isTracksAvailable || (isShowingVisibleTracks && !isVisibleTracksAvailable && recentlyVisibleGpxList.isEmpty) {
+            return localizedString("shared_string_select_all")
+        }
+        
+        return areAllTracksSelected() ? localizedString("shared_string_deselect_all") : localizedString("shared_string_select_all")
+    }
+    
+    override func getBottomButtonTitle() -> String? {
+        localizedString("shared_string_done")
+    }
+    
+    override func getTopButtonColorScheme() -> EOABaseButtonColorScheme {
+        return (!isTracksAvailable || (isShowingVisibleTracks && !isVisibleTracksAvailable && recentlyVisibleGpxList.isEmpty)) ? .inactive : .graySimple
```

Comment:
move((!isTracksAvailable || (isShowingVisibleTracks && !isVisibleTracksAvailable && recentlyVisibleGpxList.isEmpty)) to additions func (duplicate)